src/phasorpy/io.py

```diff
@@ -522,6 +540,266 @@ def read_sdt(
     return DataArray(data, **metadata)
 
 
+def read_ptu(
+    filename: str | PathLike[Any],
+    /,
+    selection: Sequence[int | slice | EllipsisType | None] | None = None,
+    *,
+    trimdims: Sequence[Literal['T', 'C', 'H']] | None = None,
+    dtype: DTypeLike | None = None,
+    frame: int | None = None,
+    channel: int | None = None,
+    keepdims: bool = True,
+) -> DataArray:
+    """Return image histogram and metadata from PicoQuant PTU T3 mode file.
+
+    PTU files contain time-correlated single photon counting measurement data
+    and instrumentation parameters.
+
+    Parameters
+    ----------
+    filename : str or Path
+        Name of PTU file to read.
+    selection : sequence of index types
+        Indices for all dimensions:
+
+        - ``None``: return all items along axis (default).
+        - ``Ellipsis``: return all items along multiple axes.
+        - ``int``: return single item along axis.
+        - ``slice``: return chunk of axis.
+          ``slice.step`` is binning factor.
+          If ``slice.step=-1``, integrate all items along axis.
+
+    trimdims : str
+        Axes to trim. The default is ``'TCH'``.
+    dtype : dtype-like
+        Unsigned integer type of image histogram array.
+        The default is ``uint16``. Increase the bit depth to avoid
+        overflows when integrating.
+    frame : int
+        If < 0, integrate time axis, else return specified frame.
+        Overrides ``selection`` for axis ``T``.
+    channel : int
+        If < 0, integrate channel axis, else return specified channel.
+        Overrides ``selection`` for axis ``C``.
+    keepdims :
+        If true (default), reduced axes are left as size-one dimension.
+
+    Returns
+    -------
+    xarray.DataArray
+        Decoded TTTR T3 records as up to 5-dimensional image array
+        with :ref:`axes codes <axes>` ``'TYXCH'`` and type specified
+        in ``dtype``:
+
+        - ``coords['H']``: times of the histogram bins.
+        - ``attrs['frequency']``: repetition frequency in MHz.
```

Comment:
The frequency returned in the files I tested was not correct, but I guess that is already wrong in the metadata of the file

Reply:
What value was returned and what did you expect? The frequency is calculated at https://github.com/cgohlke/ptufile/blob/3a2741fd3b21101d91f55ace8c55de0f5d781135/ptufile/ptufile.py#L1170-L1175. It applies to the maximum number of bins, not the actual number of bins returned. Are the `coords['H']` values correct?

Reply:
The frequency returned is 2.517700195304632 and I was told by the people who acquired the image that the frequency was 80 MHz. I don't know if the values in coords['H'], I will try to see if the problem lies there.

Reply:
I see. The "frequency" here is not the laser pulse rate but is calculated from the TCSPC resolution and the number of bins of the hardware. It's probably not useful and better to return the value of the `TTResult_SyncRate` PTU tag instead, or the inverse of `MeasDesc_GlobalResolution`. I'm not sure what to do if the time of the last bin with non-zero photons exceeds the duration between sync events...

Reply:
I changed the frequency to `TTResult_SyncRate`. Can you check again with your files?

Reply:
Now it shows the correct frequency (78.02 MHz). Thanks for improving this.